xapian-core/include/xapian/weight.h

```diff
@@ -449,20 +454,38 @@ class XAPIAN_VISIBILITY_DEFAULT TfIdfWeight : public Weight {
      *     @li 'p': Prob    idfn=log((N-Termfreq)/Termfreq)
      *     @li 'f': Freq    idfn=1/Termfreq
      *     @li 's': Squared idfn=log(N/Termfreq)^2
+     *     @li 'P': Pivoted idfn=log((N+1)/Termfreq)
      *
      * @li The third and the final character indicates the normalization for
      *     the document weight.  The following normalizations are currently
      *     supported:
      *
      *     @li 'n': None wtn=tfn*idfn
+     *     @li 'P': Pivoted wtn=wqf(tfn*idfn*(1-s+(s*normlen))+delta*idfn)
+     *              Note: Valid parameters s and delta values should be non-negative
+     *                    & non-zero when using pivoted normalization string("PPP")
      *
      * Implementing support for more normalizations of each type would require
      * extending the backend to track more statistics.
      */
     explicit TfIdfWeight(const std::string &normalizations);
 
+    explicit TfIdfWeight(double s, double delta)
+	: param_s(s), param_delta(delta)
```

Comment:
Basically, this constructor facilitates changing values of parameters `s` and `delta`. For this particular constructor to be of use normalization string should be set to `"PPP"` using another constructor:

`explicit TfIdfWeight(const std::string &normalizations);`